core/src/main/java/io/knotx/databridge/core/impl/FragmentProcessor.java

```diff
@@ -49,9 +52,23 @@ public FragmentProcessor(Vertx vertx, DataBridgeKnotOptions options) {
         .doOnNext(this::traceService)
         .flatMap(serviceEntry ->
             fetchServiceData(serviceEntry, request).toObservable()
-                .map(serviceEntry::getResultWithNamespaceAsKey))
+                .map(serviceEntry::getResultWithNamespaceAsKey)
+                .doOnError(e -> storeErrorInFragment(fragmentContext.fragment(), e, serviceEntry.getName()))
+        )
         .reduce(new JsonObject(), JsonObject::mergeIn)
-        .map(results -> applyData(fragmentContext, results));
+        .map(results -> applyData(fragmentContext, results))
+        .onErrorReturn(e -> {
+          LOGGER.error("Fragment processing failed. Cause:{}\nRequest:\n{}\nFragmentContext:\n{}\n", e.getMessage(), request.getClientRequest(), fragmentContext);
+          fragmentContext.fragment().failure(DataBridgeKnotProxy.SUPPORTED_FRAGMENT_ID, e);
+          if (!fragmentContext.fragment().fallback().isPresent()) {
```

Comment:
I think it should not happen here. Data Bridge should set error on fragment. Then at the end Assembler should check fallbacks and do some logic if the fallback is not defined.

Reply:
Data Bridge is not responsible for template evaluation, it adds the JSON object from the data source to Knot.x Context.

Reply:
The main idea behind this PR is to have a different way of processing errors for fragments, depending on Fragment configuration. 

No fallback configured -> error handling happens on Knot level. 
Fallback configured -> mark fragment as failed, error handling on Assembler level.

I added a dedicated `handleError` methot to `FragmentProcessor` to make this idea more comprehensible.